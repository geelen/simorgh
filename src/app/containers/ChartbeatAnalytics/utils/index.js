import Cookie from 'js-cookie';
import onClient from '../../../lib/utilities/onClient';
import { getPromoHeadline } from '../../../lib/analyticsUtils/article';
import { getPageTitle } from '../../../lib/analyticsUtils/frontpage';
import { getReferrer } from '../../../lib/analyticsUtils';

const ID_COOKIE = 'ckns_sylphid';

export const chartbeatUID = 50924;
export const useCanonical = true;
export const chartbeatSource = '//static.chartbeat.com/js/chartbeat.js';

const buildSectionArr = (service, value, type) => [
  `${service} - ${value}`,
  ...(type ? [`${service} - ${value} - ${type}`] : []),
];

const buildServiceType = (service, type) => [`${service} - ${type}`];

const capitalize = s => s.charAt(0).toUpperCase() + s.slice(1);

export const getSylphidCookie = () =>
  onClient() ? Cookie.get(ID_COOKIE) : null;

export const getDomain = service => {
  const serviceLower = service.toLowerCase();

  return serviceLower === 'news' ? 'bbc.co.uk' : `${service}.bbc.co.uk`;
};

export const getType = (pageType, shorthand = false) => {
  switch (pageType) {
    case 'frontPage':
    case 'index':
      return shorthand ? 'IDX' : 'Index';
    case 'article':
      return shorthand ? 'ART' : 'New Article';
    default:
      return null;
  }
};

export const buildSections = (service, type, producer, chapter) => {
  const addProducer = producer && service !== producer;
  const serviceCap = capitalize(service);
  const pageType = getType(type, true);

  const parts = [
    serviceCap,
    ...(type ? buildServiceType(serviceCap, pageType) : []),
    ...(addProducer ? buildSectionArr(serviceCap, producer, pageType) : []),
    ...(chapter ? buildSectionArr(serviceCap, chapter, pageType) : []),
  ];

  return parts.join(', ');
};

export const getTitle = (pageType, pageData, brandName) => {
  switch (pageType) {
    case 'frontPage':
    case 'index':
      return getPageTitle(pageData, brandName);
    case 'article':
      return getPromoHeadline(pageData);
    default:
      return null;
  }
};

export const getConfig = ({
  isAmp,
  platform,
  pageType,
  data,
  brandName,
  env,
  service,
  origin,
  previousPath,
}) => {
  const referrer = getReferrer(platform, origin, previousPath);
  const title = getTitle(pageType, data, brandName);
  const domain = env !== 'live' ? getDomain('test') : getDomain(service);
  const sections = buildSections(service, pageType);
  const cookie = getSylphidCookie();
  const type = getType(pageType);
  const currentPath = onClient() && window.location.pathname;
  return {
    domain,
    sections,
    uid: chartbeatUID,
    title,
    virtualReferrer: referrer,
    ...(isAmp && { contentType: type }),
    ...(!isAmp && {
      type,
      useCanonical,
      path: currentPath,
    }),
    ...(cookie && { idSync: { bbc_hid: cookie } }),
  };
};
