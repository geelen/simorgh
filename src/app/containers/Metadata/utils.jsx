import React from 'react';

export const getIconAssetUrl = (service, size) =>
  `https://news.files.bbci.co.uk/include/articles/public/${service}/images/icons/icon-${size}.png`;

const createIconLinks = (service, iconSizes, iconType) => {
  return iconSizes.map(size => {
    const iconAssetUrl = getIconAssetUrl(service, size);
    const key = `${service}-${size}`;

    if (iconType === 'icon') {
      return (
        <link
          key={key}
          rel="icon"
          type="image/png"
          href={iconAssetUrl}
          sizes={size}
        />
      );
    }

    return (
      <link key={key} rel="apple-touch-icon" sizes={size} href={iconAssetUrl} />
    );
  });
};

export const getIconLinks = (service, iconSizes) => {
  if (!iconSizes) {
    return null;
  }
  const iconTypes = Object.keys(iconSizes);
  return iconTypes.map(iconType => {
    return createIconLinks(service, iconSizes[iconType], iconType);
  });
};

export const renderAmpHtml = (ampLink, isAmp) => {
  if (isAmp) {
    return null;
  }
  return <link rel="amphtml" href={ampLink} />;
};

export const getAppleTouchUrl = service => {
  const assetsPath = process.env.SIMORGH_PUBLIC_STATIC_ASSETS_PATH || '/';
  const separatorSlash = assetsPath[assetsPath.length - 1] !== '/' ? '/' : '';

  return [
    process.env.SIMORGH_PUBLIC_STATIC_ASSETS_ORIGIN,
    assetsPath,
    separatorSlash,
    service,
    '/images/icons/icon-192x192.png',
  ].join('');
};

export const renderAlternateLinks = link => (
  <link
    rel="alternate"
    href={link.href}
    hrefLang={link.hrefLang}
    key={link.hrefLang}
  />
);
