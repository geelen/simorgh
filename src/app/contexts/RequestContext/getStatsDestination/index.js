/* Returns stats destnation for ATI based on origin, service and env
   see table on this issue https://github.com/bbc/simorgh/issues/2995
*/
const getStatsDestination = ({ isUK = true, env = 'test', service }) => {
  let destination = '';
  switch (service) {
    case 'news':
      // checks if "news" service is in the UK or not and set apprioprate destination
      destination = isUK !== false ? 'NEWS_PS' : 'NEWS_GNL';
      break;
    case 'cymrufyw':
    case 'naidheachdan':
      // checks if "cymrufyw" or "naidheachdan" service is in the UK or not and set apprioprate destination
      destination = isUK !== false ? 'NEWS_LANGUAGES_PS' : 'NEWS_LANGUAGES_GNL';
      break;
    case 'japanese':
      destination = 'NEWS_LANGUAGES_GNL';
      break;
    case 'scotland':
      destination = 'PLACEHOLDER';
      break;
    case 'archive':
      destination = 'BBC_ARCHIVE_PS';
      break;
    default:
      destination = 'WS_NEWS_LANGUAGES';
  }
  return env === 'live' ? destination : `${destination}_TEST`;
};

export default getStatsDestination;
