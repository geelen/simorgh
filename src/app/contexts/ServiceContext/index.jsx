import React from 'react';
import { node, string } from 'prop-types';
import services from '#lib/config/services/loadableConfig';
import { getVariant } from '../../lib/utilities/variantHandler';
import variantPropType from '../../models/propTypes/variants';

export const ServiceContext = React.createContext({});

export const ServiceContextProvider = ({ children, service, variant }) => {
  const LoadableContextProvider = services[service];

  if (!LoadableContextProvider) {
    return null;
  }

  return (
    <LoadableContextProvider
      Context={ServiceContext}
      dataKey={getVariant({ service, variant })}
    >
      {children}
    </LoadableContextProvider>
  );
};

ServiceContextProvider.propTypes = {
  children: node.isRequired,
  service: string,
  variant: variantPropType,
};

ServiceContextProvider.defaultProps = {
  service: 'default',
  variant: 'default',
};
