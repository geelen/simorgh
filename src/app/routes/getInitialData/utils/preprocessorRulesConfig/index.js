import applyTimestampRules from '#lib/utilities/preprocessor/rules/timestamp';
import addIdsToBlocks from '#lib/utilities/preprocessor/rules/addIdsToBlocks';
import radioPageAddIdsToBlocks from '#lib/utilities/preprocessor/rules/radioPage/addIdsToBlocks';
import applyBlockPositioning from '#lib/utilities/preprocessor/rules/blockPositioning';
import parseInternalLinks from '#lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks/blocks/internalLinks';
import convertToOptimoBlocks from '#lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks';
import addHeadlineBlock from '#lib/utilities/preprocessor/rules/cpsAssetPage/addHeadlineBlock';
import timestampToMilliseconds from '#lib/utilities/preprocessor/rules/cpsAssetPage/timestampToMilliseconds';
import filterUnknownContentTypes from '#lib/utilities/preprocessor/rules/filterContentType';
import filterEmptyGroupItems from '#lib/utilities/preprocessor/rules/filterEmptyGroupItems';
import applySquashTopstories from '#lib/utilities/preprocessor/rules/topstories';
import addIdsToItems from '#lib/utilities/preprocessor/rules/addIdsToItems';
import filterGroupsWithoutStraplines from '#lib/utilities/preprocessor/rules/filterGroupsWithoutStraplines';
import addSummaryBlock from '#lib/utilities/preprocessor/rules/cpsAssetPage/addSummaryBlock';

export const cpsAssetPreprocessorRules = [
  parseInternalLinks,
  timestampToMilliseconds,
  convertToOptimoBlocks,
  addHeadlineBlock,
  addSummaryBlock,
  applyTimestampRules,
  addIdsToBlocks,
  applyBlockPositioning,
];

export const articlesPreprocessorRules = [
  applyTimestampRules,
  addIdsToBlocks,
  applyBlockPositioning,
];

export const indexPreprocessorRules = [
  filterUnknownContentTypes,
  filterEmptyGroupItems,
  addIdsToItems,
  applySquashTopstories,
  filterGroupsWithoutStraplines,
];

export const radioPagePreprocessorRules = [radioPageAddIdsToBlocks];
