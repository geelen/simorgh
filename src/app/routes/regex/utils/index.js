const idRegex = 'c[a-zA-Z0-9]{10}o';
const ampRegex = '.amp';
const assetUriRegex = '[a-z-_]{0,}[0-9]{8,}';
const variantRegex = '/simp|/trad|/cyr|/lat';
const articleLocalRegex = 'articles|erthyglau|sgeulachdan';
const mediaIdRegex = '[a-z0-9]+';
const mediaServiceIdRegex = 'bbc_[a-z]+_radio|bbc_[a-z]+_tv';
const errorCodeRegex = '404|500';

const getServiceRegex = services => services.join('|');

export const getArticleRegex = services => {
  const serviceRegex = getServiceRegex(services);
  return `/:service(${serviceRegex})/:local(${articleLocalRegex})/:id(${idRegex}):variant(${variantRegex})?:amp(${ampRegex})?`;
};

export const getArticleSwRegex = services => {
  const serviceRegex = getServiceRegex(services);
  return `/:service(${serviceRegex})/:local(${articleLocalRegex})/sw.js`;
};

export const getArticleManifestRegex = services => {
  const serviceRegex = getServiceRegex(services);
  return `/:service(${serviceRegex})/:local(${articleLocalRegex})/manifest.json`;
};

export const getFrontPageRegex = services => {
  const serviceRegex = getServiceRegex(services);
  return `/:service(${serviceRegex}):variant(${variantRegex})?:amp(${ampRegex})?`;
};

export const getSwRegex = services => {
  const serviceRegex = getServiceRegex(services);
  return `/:service(${serviceRegex})/sw.js`;
};

export const getManifestRegex = services => {
  const serviceRegex = getServiceRegex(services);
  return `/:service(${serviceRegex})/manifest.json`;
};

export const getCpsAssetRegex = services => {
  const serviceRegex = getServiceRegex(services);
  return `/:service(${serviceRegex})/:assetUri(${assetUriRegex}):variant(${variantRegex})?:amp(${ampRegex})?`;
};

export const getRadioAndTVRegex = services => {
  const serviceRegex = getServiceRegex(services);
  return `/:service(${serviceRegex})/:serviceId(${mediaServiceIdRegex})/:mediaId(${mediaIdRegex}):amp(${ampRegex})?`;
};

export const getErrorPageRegex = services => {
  const serviceRegex = getServiceRegex(services);
  return `/:service(${serviceRegex})/:errorCode(${errorCodeRegex}):variant(${variantRegex})?`;
};
