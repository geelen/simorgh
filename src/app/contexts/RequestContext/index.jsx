import React from 'react';
import { bool, node, oneOf, string, number } from 'prop-types';
import getStatsDestination from './getStatsDestination';
import getStatsPageIdentifier from './getStatsPageIdentifier';
import getOriginContext from './getOriginContext';
import getEnv from './getEnv';
import getMetaUrls from './getMetaUrls';
import variantPropType from '../../models/propTypes/variants';

export const RequestContext = React.createContext({});

export const RequestContextProvider = ({
  children,
  bbcOrigin,
  id,
  isAmp,
  pageType,
  service,
  statusCode,
  previousPath,
  pathname,
  variant,
}) => {
  const { isUK, origin } = getOriginContext(bbcOrigin);
  const env = getEnv(origin);
  const platform = isAmp ? 'amp' : 'canonical';
  const statsDestination = getStatsDestination({
    isUK,
    env,
    service,
  });
  const statsPageIdentifier = getStatsPageIdentifier({
    pageType,
    service,
    id,
  });

  const value = {
    env,
    id,
    isUK,
    origin,
    pageType,
    isAmp,
    platform,
    statsDestination,
    statsPageIdentifier,
    statusCode,
    previousPath,
    variant,
    ...getMetaUrls(origin, pathname),
  };

  return (
    <RequestContext.Provider value={value}>{children}</RequestContext.Provider>
  );
};

RequestContextProvider.propTypes = {
  bbcOrigin: string,
  children: node.isRequired,
  id: string,
  isAmp: bool.isRequired,
  pageType: oneOf([
    'article',
    'frontPage',
    'media',
    'error',
    'MAP',
    'FIX',
    'STY',
    'PGL',
  ]).isRequired,
  service: string.isRequired,
  statusCode: number,
  pathname: string.isRequired,
  previousPath: string,
  variant: variantPropType,
};

RequestContextProvider.defaultProps = {
  bbcOrigin: null,
  id: null,
  statusCode: null,
  previousPath: null,
  variant: null,
};
