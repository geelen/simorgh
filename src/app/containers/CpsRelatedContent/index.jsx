import React, { useContext } from 'react';
import { arrayOf, shape } from 'prop-types';
import SectionLabel from '@bbc/psammead-section-label';
import styled from 'styled-components';
import { StoryPromoLi, StoryPromoUl } from '@bbc/psammead-story-promo-list';
import path from 'ramda/src/path';
import assocPath from 'ramda/src/assocPath';

import { GEL_GROUP_3_SCREEN_WIDTH_MIN } from '@bbc/gel-foundations/breakpoints';
import {
  GEL_SPACING_DBL,
  GEL_SPACING_TRPL,
} from '@bbc/gel-foundations/spacings';

import { RequestContext } from '#contexts/RequestContext';
import { storyItem } from '#models/propTypes/storyItem';
import { ServiceContext } from '#contexts/ServiceContext';
import { GhostGrid, GridItemConstrainedLarge } from '#lib/styledGrid';
import StoryPromo from '../StoryPromo';

const Wrapper = styled(GridItemConstrainedLarge)`
  margin-bottom: ${GEL_SPACING_DBL};
  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
    margin-bottom: ${GEL_SPACING_TRPL};
  }

  /* z-index needs explicitly set as the psammead section label component uses negative z-indices */
  z-index: 0;
`;

const formatItem = (item, env) => {
  if (env === 'live') return item;

  // In non-live environments, we need to pass this querystring to ensure
  // the linked site retrieves its data from the TEST API location
  const uriSuffix = '?_x_candy_override=https%3A%2F%2Fapi.test.bbc.co.uk';
  const baseUri = path(['locators', 'assetUri'], item);

  return assocPath(['locators', 'assetUri'], `${baseUri}${uriSuffix}`, item);
};

const CpsRelatedContent = ({ content }) => {
  const { script, service, dir, translations } = useContext(ServiceContext);
  const { env } = useContext(RequestContext);
  if (!content.length) return null;

  return (
    <GhostGrid
      as="aside"
      role="complementary"
      aria-labelledby="related-content-heading"
    >
      <Wrapper>
        <SectionLabel
          script={script}
          service={service}
          dir={dir}
          labelId="related-content-heading"
        >
          {translations.relatedContent}
        </SectionLabel>

        <StoryPromoUl>
          {content
            .map(item => formatItem(item, env))
            .map(item => (
              <StoryPromoLi key={item.id || item.uri}>
                <StoryPromo item={item} />
              </StoryPromoLi>
            ))}
        </StoryPromoUl>
      </Wrapper>
    </GhostGrid>
  );
};

CpsRelatedContent.propTypes = {
  // We Reuse the front page story item blocks
  // Both pages use CPS, so the data schema is the same
  // This can be found under CPS ARES payloads: relatedContent.groups[0].promos
  content: arrayOf(shape(storyItem)),
};

CpsRelatedContent.defaultProps = {
  content: [],
};

export default CpsRelatedContent;
