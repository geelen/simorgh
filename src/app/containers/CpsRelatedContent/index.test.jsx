import React from 'react';
import { render } from '@testing-library/react';
import path from 'ramda/src/path';

import { ServiceContextProvider } from '#contexts/ServiceContext';
import { RequestContextProvider } from '#contexts/RequestContext';

import CpsRelatedContent from '.';
import pidginPageData from '#data/pidgin/cpsAssets/tori-49450859';

import preprocessor from '#lib/utilities/preprocessor';
import { cpsAssetPreprocessorRules } from '#app/routes/getInitialData/utils/preprocessorRulesConfig';

const promos = path(['relatedContent', 'groups', 0, 'promos'], pidginPageData);

// eslint-disable-next-line react/prop-types
const renderRelatedContent = ({
  content = promos,
  bbcOrigin = 'https://www.test.bbc.co.uk',
} = {}) => {
  return render(
    <ServiceContextProvider service="pidgin">
      <RequestContextProvider
        bbcOrigin={bbcOrigin}
        isAmp={false}
        pageType="MAP"
        pathname="/pidgin/tori-49450859"
        service="pidgin"
        statusCode={200}
      >
        <CpsRelatedContent content={content} />
      </RequestContextProvider>
    </ServiceContextProvider>,
  );
};

describe('CpsRelatedContent', () => {
  it('should render Story Promo components when given appropriate data', () => {
    // Ensure fixture still has promos
    expect(promos.length).toBe(3);

    const { asFragment } = renderRelatedContent();

    expect(document.querySelectorAll(`li[class^='StoryPromoLi']`).length).toBe(
      promos.length,
    );

    // All related content links should have the x_candy_override querystring in the test environment
    expect(document.querySelectorAll(`[href*='x_candy_override']`).length).toBe(
      promos.length,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render Story Promo components in Live environment', () => {
    const { asFragment } = renderRelatedContent({
      bbcOrigin: 'https://www.bbc.co.uk',
    });

    // x_candy_override should not be used in the live environment
    expect(document.querySelector(`[href*='x_candy_override']`)).toBeNull();
    expect(asFragment()).toMatchSnapshot();
  });

  it('should have a "complementary" role (a11y)', () => {
    renderRelatedContent();
    expect(document.querySelectorAll(`[role='complementary']`).length).toBe(1);
  });

  it('should render timestamps in milliseconds when page data has timestamps in seconds', async () => {
    const initialPromo = [
      {
        ...promos[0],
        timestamp: 1234567890,
      },
    ];

    const pageData = await preprocessor(
      {
        ...pidginPageData,
        relatedContent: { groups: [{ promos: initialPromo }] },
      },
      cpsAssetPreprocessorRules,
    );

    const transformedPromos = path(
      ['relatedContent', 'groups', 0, 'promos'],
      pageData,
    );

    const { getByText } = renderRelatedContent({
      content: transformedPromos,
    });

    expect(getByText('February 2009', { exact: false })).not.toBeNull();
  });
});
