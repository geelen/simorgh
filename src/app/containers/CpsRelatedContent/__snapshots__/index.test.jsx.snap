// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CpsRelatedContent should render Story Promo components in Live environment 1`] = `
<DocumentFragment>
  <aside
    aria-labelledby="related-content-heading"
    class="GhostGrid-sc-12lwanc-1 gfRQD"
    role="complementary"
  >
    <div
      class="GridItemConstrainedLarge-sc-12lwanc-5 Wrapper-sc-1itct8j-0 ebGqUw"
    >
      <div
        class="SectionLabelWrapper-sc-1x4gwjz-1 fKCwVP"
      >
        <div
          class="Bar-sc-1x4gwjz-0 qaAlR"
        />
        <h2
          class="Heading-sc-1x4gwjz-2 eZofTG"
        >
          <div
            class="FlexColumn-wx5kdp-0 dgpsiS"
          >
            <span
              class="FlexRow-wx5kdp-2 dVBNyZ"
            >
              <span
                class="Title-wx5kdp-4 jiurmA"
                dir="ltr"
                id="related-content-heading"
              >
                Another thing we de for inside dis tori
              </span>
            </span>
          </div>
        </h2>
      </div>
      <ul
        class="StoryPromoUl-sc-171lqjd-1 isCYWc"
        role="list"
      >
        <li
          class="StoryPromoLi-sc-171lqjd-0 diwbBz"
          role="listitem"
        >
          <div
            class="StoryPromoWrapper-sc-1dvfmi3-0 eYdwfL"
          >
            <div
              class="ImageGridItem-sc-1dvfmi3-1 diQWZw"
            >
              <div
                class="ImageContentsWrapper-sc-1dvfmi3-2 hEkcUX"
              >
                <div
                  class="ImagePlaceholder-sc-11u25v2-0 fNqlUz"
                >
                  <div
                    class="lazyload-placeholder"
                  />
                </div>
                <div
                  class="InlineMediaIndicator-sc-1dvfmi3-3 ivGVpf"
                />
              </div>
            </div>
            <div
              class="TextGridItem-sc-1dvfmi3-6 fdmIIY"
            >
              <h3
                class="Headline-sc-1dvfmi3-4 cWuUJx"
              >
                <a
                  class="Link-sc-1dvfmi3-7 iSHpco"
                  href="/pidgin/44508901"
                >
                  <span
                    role="text"
                  >
                    <span
                      class="VisuallyHiddenText-sc-1yjwwa5-0 jFfEvX"
                    >
                      , 
                    </span>
                    <span>
                      Meet boys who dey convert cassava to electricity
                    </span>
                  </span>
                </a>
              </h3>
              <p
                class="Summary-sc-1dvfmi3-5 itgvLR"
              >
                James den Kwesi Ansah want convert most of Ghana's cassava waste into cheap electricity supply for schools, hospitals den villages.
              </p>
              <time
                class="StyledTimestamp-um718p-0 kVWTyt"
                datetime="1970-01-18"
              >
                18th January 1970
              </time>
            </div>
          </div>
        </li>
        <li
          class="StoryPromoLi-sc-171lqjd-0 diwbBz"
          role="listitem"
        >
          <div
            class="StoryPromoWrapper-sc-1dvfmi3-0 eYdwfL"
          >
            <div
              class="ImageGridItem-sc-1dvfmi3-1 diQWZw"
            >
              <div
                class="ImageContentsWrapper-sc-1dvfmi3-2 hEkcUX"
              >
                <div
                  class="ImagePlaceholder-sc-11u25v2-0 fNqlUz"
                >
                  <div
                    class="lazyload-placeholder"
                  />
                </div>
                <div
                  class="InlineMediaIndicator-sc-1dvfmi3-3 ivGVpf"
                />
              </div>
            </div>
            <div
              class="TextGridItem-sc-1dvfmi3-6 fdmIIY"
            >
              <h3
                class="Headline-sc-1dvfmi3-4 cWuUJx"
              >
                <a
                  class="Link-sc-1dvfmi3-7 iSHpco"
                  href="/pidgin/tori-46975713"
                >
                  How light companies dey use estimated billing show Nigerians pepper
                </a>
              </h3>
              <p
                class="Summary-sc-1dvfmi3-5 itgvLR"
              >
                Lawmakers for Nigeria wan make am crime for light companies to use guesswork give customer estimated bill.
              </p>
              <time
                class="StyledTimestamp-um718p-0 kVWTyt"
                datetime="1970-01-18"
              >
                18th January 1970
              </time>
            </div>
          </div>
        </li>
        <li
          class="StoryPromoLi-sc-171lqjd-0 diwbBz"
          role="listitem"
        >
          <div
            class="StoryPromoWrapper-sc-1dvfmi3-0 eYdwfL"
          >
            <div
              class="ImageGridItem-sc-1dvfmi3-1 diQWZw"
            >
              <div
                class="ImageContentsWrapper-sc-1dvfmi3-2 hEkcUX"
              >
                <div
                  class="ImagePlaceholder-sc-11u25v2-0 fNqlUz"
                >
                  <div
                    class="lazyload-placeholder"
                  />
                </div>
                <div
                  class="InlineMediaIndicator-sc-1dvfmi3-3 ivGVpf"
                />
              </div>
            </div>
            <div
              class="TextGridItem-sc-1dvfmi3-6 fdmIIY"
            >
              <h3
                class="Headline-sc-1dvfmi3-4 cWuUJx"
              >
                <a
                  class="Link-sc-1dvfmi3-7 iSHpco"
                  href="/pidgin/tori-42494678"
                >
                  Nigeria: Wetin 5,222 megawatts electric fit do?
                </a>
              </h3>
              <p
                class="Summary-sc-1dvfmi3-5 itgvLR"
              >
                Di Transmission Company of Nigeria and di vice-president office carry tori come outside say electric power wey di country produce don increase pass before as e reach 5,222 megawatts dis December.
              </p>
              <time
                class="StyledTimestamp-um718p-0 kVWTyt"
                datetime="1970-01-18"
              >
                18th January 1970
              </time>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </aside>
</DocumentFragment>
`;

exports[`CpsRelatedContent should render Story Promo components when given appropriate data 1`] = `
<DocumentFragment>
  <aside
    aria-labelledby="related-content-heading"
    class="GhostGrid-sc-12lwanc-1 gfRQD"
    role="complementary"
  >
    <div
      class="GridItemConstrainedLarge-sc-12lwanc-5 Wrapper-sc-1itct8j-0 ebGqUw"
    >
      <div
        class="SectionLabelWrapper-sc-1x4gwjz-1 fKCwVP"
      >
        <div
          class="Bar-sc-1x4gwjz-0 qaAlR"
        />
        <h2
          class="Heading-sc-1x4gwjz-2 eZofTG"
        >
          <div
            class="FlexColumn-wx5kdp-0 dgpsiS"
          >
            <span
              class="FlexRow-wx5kdp-2 dVBNyZ"
            >
              <span
                class="Title-wx5kdp-4 jiurmA"
                dir="ltr"
                id="related-content-heading"
              >
                Another thing we de for inside dis tori
              </span>
            </span>
          </div>
        </h2>
      </div>
      <ul
        class="StoryPromoUl-sc-171lqjd-1 isCYWc"
        role="list"
      >
        <li
          class="StoryPromoLi-sc-171lqjd-0 diwbBz"
          role="listitem"
        >
          <div
            class="StoryPromoWrapper-sc-1dvfmi3-0 eYdwfL"
          >
            <div
              class="ImageGridItem-sc-1dvfmi3-1 diQWZw"
            >
              <div
                class="ImageContentsWrapper-sc-1dvfmi3-2 hEkcUX"
              >
                <div
                  class="ImagePlaceholder-sc-11u25v2-0 fNqlUz"
                >
                  <div
                    class="lazyload-placeholder"
                  />
                </div>
                <div
                  class="InlineMediaIndicator-sc-1dvfmi3-3 ivGVpf"
                />
              </div>
            </div>
            <div
              class="TextGridItem-sc-1dvfmi3-6 fdmIIY"
            >
              <h3
                class="Headline-sc-1dvfmi3-4 cWuUJx"
              >
                <a
                  class="Link-sc-1dvfmi3-7 iSHpco"
                  href="/pidgin/44508901?_x_candy_override=https%3A%2F%2Fapi.test.bbc.co.uk"
                >
                  <span
                    role="text"
                  >
                    <span
                      class="VisuallyHiddenText-sc-1yjwwa5-0 jFfEvX"
                    >
                      , 
                    </span>
                    <span>
                      Meet boys who dey convert cassava to electricity
                    </span>
                  </span>
                </a>
              </h3>
              <p
                class="Summary-sc-1dvfmi3-5 itgvLR"
              >
                James den Kwesi Ansah want convert most of Ghana's cassava waste into cheap electricity supply for schools, hospitals den villages.
              </p>
              <time
                class="StyledTimestamp-um718p-0 kVWTyt"
                datetime="1970-01-18"
              >
                18th January 1970
              </time>
            </div>
          </div>
        </li>
        <li
          class="StoryPromoLi-sc-171lqjd-0 diwbBz"
          role="listitem"
        >
          <div
            class="StoryPromoWrapper-sc-1dvfmi3-0 eYdwfL"
          >
            <div
              class="ImageGridItem-sc-1dvfmi3-1 diQWZw"
            >
              <div
                class="ImageContentsWrapper-sc-1dvfmi3-2 hEkcUX"
              >
                <div
                  class="ImagePlaceholder-sc-11u25v2-0 fNqlUz"
                >
                  <div
                    class="lazyload-placeholder"
                  />
                </div>
                <div
                  class="InlineMediaIndicator-sc-1dvfmi3-3 ivGVpf"
                />
              </div>
            </div>
            <div
              class="TextGridItem-sc-1dvfmi3-6 fdmIIY"
            >
              <h3
                class="Headline-sc-1dvfmi3-4 cWuUJx"
              >
                <a
                  class="Link-sc-1dvfmi3-7 iSHpco"
                  href="/pidgin/tori-46975713?_x_candy_override=https%3A%2F%2Fapi.test.bbc.co.uk"
                >
                  How light companies dey use estimated billing show Nigerians pepper
                </a>
              </h3>
              <p
                class="Summary-sc-1dvfmi3-5 itgvLR"
              >
                Lawmakers for Nigeria wan make am crime for light companies to use guesswork give customer estimated bill.
              </p>
              <time
                class="StyledTimestamp-um718p-0 kVWTyt"
                datetime="1970-01-18"
              >
                18th January 1970
              </time>
            </div>
          </div>
        </li>
        <li
          class="StoryPromoLi-sc-171lqjd-0 diwbBz"
          role="listitem"
        >
          <div
            class="StoryPromoWrapper-sc-1dvfmi3-0 eYdwfL"
          >
            <div
              class="ImageGridItem-sc-1dvfmi3-1 diQWZw"
            >
              <div
                class="ImageContentsWrapper-sc-1dvfmi3-2 hEkcUX"
              >
                <div
                  class="ImagePlaceholder-sc-11u25v2-0 fNqlUz"
                >
                  <div
                    class="lazyload-placeholder"
                  />
                </div>
                <div
                  class="InlineMediaIndicator-sc-1dvfmi3-3 ivGVpf"
                />
              </div>
            </div>
            <div
              class="TextGridItem-sc-1dvfmi3-6 fdmIIY"
            >
              <h3
                class="Headline-sc-1dvfmi3-4 cWuUJx"
              >
                <a
                  class="Link-sc-1dvfmi3-7 iSHpco"
                  href="/pidgin/tori-42494678?_x_candy_override=https%3A%2F%2Fapi.test.bbc.co.uk"
                >
                  Nigeria: Wetin 5,222 megawatts electric fit do?
                </a>
              </h3>
              <p
                class="Summary-sc-1dvfmi3-5 itgvLR"
              >
                Di Transmission Company of Nigeria and di vice-president office carry tori come outside say electric power wey di country produce don increase pass before as e reach 5,222 megawatts dis December.
              </p>
              <time
                class="StyledTimestamp-um718p-0 kVWTyt"
                datetime="1970-01-18"
              >
                18th January 1970
              </time>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </aside>
</DocumentFragment>
`;
